### PR TITLE
JIRA issues 442, 446, 447, 448

### PR DESCRIPTION
**Pull request by PaulBoon**
# RFI Checklist

_**Before** submitting the pull request, fill out sections (1.) Related Issues and (2.) Pull Request Checklist._

### 1. Related Issues

_List and [link](https://guides.github.com/features/issues/#notifications) to the issues in this Pull Request._

- [#### + Title HERE]

---
### 2. Pull Request Checklist

- [ ]  Functionality completed as described in FRD
- [ ]  Dependencies, risks, assumptions in FRD addressed
- [ ]  Unit tests completed
- [ ]  Deployment requirements identified (e.g., SQL scripts, indexing)
- [ ]  Documentation completed
- [ ]  All code checkins completed

---
### 3. Review Checklist

_**After** the pull request has been submitted, fill out this section._

- [ ]  Code review completed or waived
- [ ]  Testing requirements completed
- [ ]  Usability testing completed or waived
- [ ]  Support testing completed or waived
- [ ]  Merged with develop branch and resolved conflicts

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/DataverseBridgeDialog.java [hunk 1]
```diff
@@ -52,18 +50,32 @@
...
 
     @PostConstruct
     public void init() {
         if (session.getUser().isAuthenticated() && settingsService.getValueForKey(SettingsServiceBean.Key.DataverseBridgeConf) != null) {
             dataverseBridge =  new DataverseBridge(((AuthenticatedUser) session.getUser()).getEmail(), settingsService, datasetService, datasetVersionService, authService, mailServiceBean);
-            dvDarConfs = dataverseBridge.getDvBridgeConf().getConf();
-            darNames = new ArrayList<>(dvDarConfs.keySet());
+            dataverseBridgeSetting = dataverseBridge.getDataverseBridgeSetting();
+            darNameList= dataverseBridgeSetting.getDarNames();
+            if (darNameList.size() > 1) {
```
Comment (PaulBoon) [inline src/main/java/edu/harvard/iq/dataverse/DataverseBridgeDialog.java]: What if the list has only one entry? 
Are you sure it is not '>='.

File: src/main/java/edu/harvard/iq/dataverse/DataverseBridgeDialog.java [hunk 2]
```diff
@@ -73,67 +85,54 @@ public void archive() {
...
+            logger.severe(e.getMessage());
+            state = DataverseBridge.StateEnum.INTERNAL_SERVER_ERROR;
+        }
+        if (state == DataverseBridge.StateEnum.IN_PROGRESS) {
+            dataverseBridge.updateDataverseVersionState(persistentId, datasetVersionFriendlyNumber, state.toString() + "@" + darName);
+            Flowable.fromCallable(() -> {
+                DataverseBridge.StateEnum currentState = DataverseBridge.StateEnum.IN_PROGRESS;
+                int hopCount = 0;
+                while ((currentState == DataverseBridge.StateEnum.IN_PROGRESS) && hopCount < 10) {
+                    Thread.sleep(900000);//15 minutes
```
Comment (PaulBoon) [inline src/main/java/edu/harvard/iq/dataverse/DataverseBridgeDialog.java]: Could you make this 'wait time' and 'max hop count' configurable

File: src/main/java/edu/harvard/iq/dataverse/DataverseBridgeDialog.java [hunk 3]
```diff
@@ -73,67 +85,54 @@ public void archive() {
...
+            })
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(Schedulers.single())
+                    .doOnError(
+                            throwable -> {
+                                String es = throwable.toString();
+                                logger.severe(es);
+                            }
+                    )
+                    .subscribe(cs -> logger.info("The dataset of '" + persistentId + "' has been archived."),
```
Comment (PaulBoon) [inline src/main/java/edu/harvard/iq/dataverse/DataverseBridgeDialog.java]: You don't log the state, which can be IN_PROGRESS if we reached 10 hops...

File: src/main/webapp/dataset.xhtml [hunk 4]
```diff
@@ -243,16 +243,21 @@
...
                                                					and DatasetPage.dataset.owner.released and permissionsWrapper.canIssuePublishDatasetCommand(DatasetPage.dataset)}">
                                     <f:setPropertyActionListener target="#{dataverseBridgeDialog.datasetVersionFriendlyNumber}" value="#{DatasetPage.workingVersion.versionNumber}.#{DatasetPage.workingVersion.minorVersionNumber}"/>
                                     <f:setPropertyActionListener target="#{dataverseBridgeDialog.persistentId}" value="#{DatasetPage.persistentId}"/>
-
+                                    <f:setPropertyActionListener target="#{dataverseBridgeDialog.darName}" value="#{DatasetPage.darName}"/>
+                                    <f:setPropertyActionListener target="#{dataverseBridgeDialog.darNameList}" value="#{DatasetPage.darNameList}"/>
+                                    <f:setPropertyActionListener target="#{dataverseBridgeDialog.darUsername}" value="#{DatasetPage.darUsername}"/>
+                                    <f:setPropertyActionListener target="#{dataverseBridgeDialog.darPassword}" value="#{DatasetPage.darPassword}"/>
```
Comment (PaulBoon) [inline src/main/webapp/dataset.xhtml]: Why is the password in here, it's not an input from the user is it?

File: src/main/webapp/dataverseBridgeFragment.xhtml [hunk 5]
```diff
@@ -9,29 +9,39 @@
...
+                                     requiredMessage="#{bundle['dataset.archive.dialog.required.username']}" required="true"/>
+                        <h:message for="darUsername" styleClass="bg-danger text-danger"/>
 
-                    <p:outputLabel for="password" value="#{bundle['dataset.archive.dialog.label.password']}"/>
-                    <p:password id="password" value="#{dataverseBridgeDialog.darPassword}"
-                                requiredMessage="#{bundle['dataset.archive.dialog.required.password']}" required="true"/>
-                    <h:message for="password" styleClass="bg-danger text-danger"/>
-                </h:panelGrid>
+                        <p:outputLabel for="password" value="#{bundle['dataset.archive.dialog.label.password']}"/>
+                        <p:password id="password" value="#{dataverseBridgeDialog.darPassword}"
```
Comment (PaulBoon) [inline src/main/webapp/dataverseBridgeFragment.xhtml]: Here again a password, why?

File: src/main/java/edu/harvard/iq/dataverse/DatasetPage.java [hunk 6]
```diff
@@ -1555,24 +1554,44 @@ private String init(boolean initFull) {
...
             if (!isUserHasAdminRole(rs))
                 return null;
+            darNameList = new ArrayList<>();
             DataverseBridge dbd = new DataverseBridge(((AuthenticatedUser) session.getUser()).getEmail(), settingsService, datasetService, datasetVersionService, authService, mailServiceBean);
-            DataverseBridge.DvBridgeConf dvBridgeConf = dbd.getDvBridgeConf();
-            dataverseBridgeEnabled = isUserBelongsToSwordGroup(rs, dvBridgeConf.getUserGroup());
+            DataverseBridge.DataverseBridgeSetting dataverseBridgeSetting = dbd.getDataverseBridgeSetting();
+            List<DataverseBridge.DarSetting> darSettingList = dataverseBridgeSetting.getDarSettings();
+            if (darSettingList.isEmpty()) {
+                //todo: errror message
```
[the remark this answers is not in the document]
Comment (ekoi) [reply src/main/java/edu/harvard/iq/dataverse/DatasetPage.java]: done!

File: src/main/java/edu/harvard/iq/dataverse/DataverseBridge.java [hunk 7]
```diff
@@ -167,64 +169,74 @@ public StateEnum checkArchivingProgress(String dvBaseMetadataXml, String persist
...
-            mailServiceBean.sendSystemEmail(authService.getAuthenticatedUser("dataverseAdmin").getEmail()
-                    , state.value, "persistentId: " + persistentId + "\nVersion: " + datasetVersionFriendlyNumber );
+        addMessage(fs, displayMessage, displayMessageDetails);
+        if (sendMailToAdmin) {
+            mailServiceBean.sendSystemEmail(authService.getAuthenticatedUser("dataverseAdmin").getEmail(), emailSubject, emailMessage);
+        }
+    }
+
+    public String getDataverseVersionNoteText(String persistentId, String datasetVersionFriendlyNumber) {
+        Dataset dataset = datasetService.findByGlobalId(persistentId);
```
[the remark this answers is not in the document]
Comment (ekoi) [reply src/main/java/edu/harvard/iq/dataverse/DataverseBridge.java]: done.

File: src/main/webapp/dataset.xhtml (hunk 4 above)
Comment (PaulBoon) [reply src/main/webapp/dataset.xhtml]: Ok, it is configured at the Dataverse app

File: src/main/webapp/dataverseBridgeFragment.xhtml (hunk 5 above)
Comment (PaulBoon) [reply src/main/webapp/dataverseBridgeFragment.xhtml]: Ok, same as above

**Pull request merged**